third_party/terraform/utils/common_operation.go

```diff
@@ -90,35 +92,22 @@ func OperationDone(w Waiter) bool {
 
 func CommonRefreshFunc(w Waiter) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		// First, read the operation from the server.
 		op, err := w.QueryOp()
-
-		// If we got a non-retryable error, return it.
 		if err != nil {
-			if !isRetryableError(err) {
```

Comment:
Why would you not retry requests for the Operation? Getting a 502 against an Operation request shouldn't error out the CRUD operation.

Reply:
From PR body: 

> Generated ops have retries built in, and handwritten clients need to opt-in. I couldn't find any other cases of failing ops, and we'll be using entirely generated ops sooner rather than later so I only did SQL; if more APIs need it we can opt them in too but trawling our test logs I didn't see any.

From comment added in this PR:

https://github.com/GoogleCloudPlatform/magic-modules/pull/1561/files#diff-ba94bd6cc206d3df5971ffc843f8936bR26

Reply:
Why wouldn't centralize the retry logic for operations rather than force handwritten resources to opt in? I understand that eventually it wouldn't be needed but having cascading retries isn't bad and keeping Operation error handling next to the common operation request logic makes sense. Not handing the retry logic is a failure scenario and making a method consumer handle internal failure scenarios seems like an anti-pattern.

Reply:
Separation of concerns imo- a product-specific operation should know how to query the product API for an operation object and the common operation waiter only needs to know how to query an operation. So in that model, API-specific rate limits are something handled at the operation level.

It's also worth noting that this didn't work for handwritten operations at all before because their clients guarantee that their return exactly one non-nil field, so we can be assured that this is only required by SQL (so far)

Reply:
I think that providing an interface where you need to know the common failure scenarios and account for them isn't a great experience. But I also see that the transport mechanism for fetching the Operation isn't part of the interface. So even though everybody is using HTTP and having to copy/paste HTTP error handling code sucks, handling those http errors in CommonRefresh is actually a leaky abstraction. You've convinced me 👍